pkg/resource/observer_task.go

```diff
@@ -634,7 +635,7 @@ func (m *OBServerManager) WaitOBServerDeletedInCluster() error {
 			return errors.Wrapf(err, "Get oceanbase operation manager failed")
 		}
 		observer, err := operationManager.GetServer(observerInfo)
-		if observer == nil {
+		if observer == nil && err == nil {
```

Comment:
What if `err != nil`, shall we log the error information?

Reply:
ok, I'll add log for error